dashboard/app/views/home/_homepage.html.haml

```diff
@@ -14,19 +14,37 @@
     - homepage_data[:isTeacher] = true
     - homepage_data[:joined_sections] = @student_sections
     - homepage_data[:announcement] = DCDO.get('announcement_override', nil)
-    - show_census_banner = !!(current_user.show_census_teacher_banner?)
+    - if !current_user.donor_teacher_banner_dismissed
+      - donor_banner_name = current_user.donor_teacher_banner_name
+    - else
+      - donor_banner_name = @force_donor_teacher_banner
+    - show_census_banner = !!(!donor_banner_name && current_user.show_census_teacher_banner?)
```

Comment:
It might be time to extract this banner selection logic to a view helper.

Reply:
Yes, I'd like to do a followup PR which hoists all of this to the controller.

Reply:
Prepped in https://github.com/code-dot-org/code-dot-org/pull/30753.